### DISCUSSION TITLE
What is pool storage resource and other stuff

### DISCUSSION THREAD

**Discussion by dkoci** [noobaa/noobaa-core#6840] [Q&A] [answered]
Hi.
Can someone please explain me what is POOL STORAGE RESOURCE and when should we use it?

![68317694-9B56-47D5-8500-E42E28E0A8BE](https://user-images.githubusercontent.com/5565897/147164628-1684090b-f284-42b0-8291-c90f80be819f.jpeg)

I have read alot about noobaa, i have watched videos about nooba on yt but I am still confused. 
Namespace type of resources is another thing which I dont understand.
Lets say we have a scenario: connect to a private s3 storage and create a bucket. Which resource type would be the right one (pool, namespace...)?
Is storage resource (web cobsole) and backing-store (noobaa cli) the same thing?  Do we have to ceeate bucket class too (i dont find bucket class create option in web console, but I noticed it in some yt video using noobaa cli).

Thanks for help

**Comment by nimrod-becker** [comment, marked as answer]
Pool storage resource is the equivilant to the pv pool BackingStore, if you looked at the operator and the different types of BackingStores/Resources (two names, same entity, as you suspected). It means storing the data on top of PVs attached to pods in the same cluster NooBaa is running.

Regarding BucketClass, you can look at that as a template for a new bucket. When you create a new bucket via OBC (BucketClass does not take effect when you use the S3 API itself to create a new bucket) the different setting on the BucketClass (if provided during the OBC creation) will be set on the new bucket. Since its only a template, things can be either overridden on the OBC yaml itself or later on. See [BucketClass CRD](https://github.com/noobaa/noobaa-operator/blob/master/doc/bucket-class-crd.md) for more info.

The difference between a non-namespace resource and a namespace resource is simply but what type of bucket you want to attach it to. non-namespace buckets use non-namespace resources and namespace buckets use namespace resources. Now, the difference between a namespace bucket and a non-namespace bucket is the following:

For a non-namespace bucket, NooBaa will take the stream of the object that is being uploaded, will Dedup it (inline), compress each chunk which is the result of the dedup, and then encrypt each chunk. Then these chinks are being written on the resource itself. In this scenario you gain dedup and compression, and also the additional protection of each chunk (averaged around 4MB) being encrypted. However, in order to access the data you would have to go via NooBaa.

A namespace bucket behaves differently, in this case, NooBaa will write the data as received by the client. This means we won't alter the data by dedup, compression, or encryption so we won't benefit from those. However, since the data was written "as is" it can also be accessed directly from the resource without going via NooBaa. 

So it really depends on our need and use case

**Reply by dkoci** [reply]
Thank you for the explanation.
Is it possible to delete default backingstore which was created by the operator? I am unable to delete it. Default OBC which was created by the operator uses default storage class (which is some nfs storage class - default storage class in my cluster) instead of nooba storage class.

**Comment by nimrod-becker** [comment]
Right now it is not, you can create new BackingStores and use them and not the default, but currently deleting it would only cause a reconcile.

It is in our future plans to be enable switching the default into another one created by the user

**Reply by guymguym** [reply]
The operator will create the default bucket class if it doesn't exist, but it won't override it, so I think @dkoci should be able to edit the default bucket class and change it as desired.

**Reply by dkoci** [reply]
The case is in the STORAGE class, not bucket class. I dont understand why default storage class created by the operator uses my default provisioner which is some nfs provisioner. It means that data will be stored in NFS instead of s3 storage? I really dont get it when should we use pool storage.
There is also a PVC created by the operator which uses the default noobaa storage class. Does thins PVC store the data in NFS?
